ambry-rest/src/test/java/com.github.ambry.rest/NettyServerFactoryTest.java

```diff
@@ -30,74 +36,68 @@
+      } else {
+        assertEquals("Expected one ChannelInitializer when SSLFactoryImpl is null", 1, channelInitializers.size());
+      }
+      assertNotNull("No ChannelInitializer for plaintext port", channelInitializers.get(nettyConfig.nettyServerPort));
+    }
   }
 
   /**
    * Tests instantiation of {@link NettyServerFactory} with bad input.
    */
   @Test
-  public void getNettyServerFactoryWithBadInputTest() {
+  public void getNettyServerFactoryWithBadInputTest() throws Exception {
     // dud properties. server should pick up defaults
     Properties properties = new Properties();
     VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
     MetricRegistry metricRegistry = new MetricRegistry();
     RestRequestHandler restRequestHandler = new MockRestRequestResponseHandler();
     PublicAccessLogger publicAccessLogger = new PublicAccessLogger(new String[]{}, new String[]{});
     RestServerState restServerState = new RestServerState("/healthCheck");
+    SSLFactory sslFactory = new MockSSLFactory(verifiableProperties);
 
-    // VerifiableProperties null.
-    try {
-      new NettyServerFactory(null, metricRegistry, restRequestHandler, publicAccessLogger, restServerState);
-      fail("Instantiation should have failed because one of the arguments was null");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
-
-    // MetricRegistry null.
-    try {
-      new NettyServerFactory(verifiableProperties, null, restRequestHandler, publicAccessLogger, restServerState);
-      fail("Instantiation should have failed because one of the arguments was null");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
-
-    // RestRequestHandler null.
-    try {
-      new NettyServerFactory(verifiableProperties, metricRegistry, null, publicAccessLogger, restServerState);
-      fail("Instantiation should have failed because one of the arguments was null");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
-
-    // PublicAccessLogger null.
-    try {
-      new NettyServerFactory(verifiableProperties, metricRegistry, restRequestHandler, null, restServerState);
-      fail("Instantiation should have failed because one of the arguments was null");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
+    for (int i = 0; i < 5; i++) {
+      try {
+        new NettyServerFactory(i == 0 ? verifiableProperties : null, i == 1 ? metricRegistry : null,
+            i == 2 ? restRequestHandler : null, i == 3 ? publicAccessLogger : null, i == 4 ? restServerState : null,
+            sslFactory);
+        fail("Instantiation should have failed because argument " + i + " was null");
+      } catch (IllegalArgumentException e) {
+        // expected. Nothing to do.
+      }
     }
 
-    // RestServerState null.
+    // Should be int. So will throw at instantiation.
```

Comment:
Not sure if this test belongs here since the `int` is enforced by `NettyConfig` and not by `NettyServerFactory`. I noticed that you have removed the equivalent plain text port text in `NettyServer` (which is a good move I think).